src/main/java/dk/sdu/se_f22/sortingmodule/scoring/Scoring.java

```diff
@@ -120,14 +118,14 @@ public List<Product> unWrapProduct (List<ProductScore> input) {
     }
 
     @Override
-    public SearchHits scoreSort(SearchHits input, String type) {
+    public SearchHits scoreSort(SearchHits input, ScoreSortType type) {
 
         switch (type) {
-            case "all" -> input.setProducts(scoreSortAll(input.getProducts()));
-            case "price" -> input.setProducts(scoreSortPrice(input.getProducts()));
-            case "review" -> input.setProducts(scoreSortReview(input.getProducts()));
-            case "stock" -> input.setProducts(scoreSortStock(input.getProducts()));
-            case "date" -> input.setProducts(scoreSortDate(input.getProducts()));
+            case ALL -> input.setProducts(scoreSortAll(input.getProducts()));
+            case PRICE -> input.setProducts(scoreSortPrice(input.getProducts()));
+            case REVIEW -> input.setProducts(scoreSortReview(input.getProducts()));
+            case STOCK -> input.setProducts(scoreSortStock(input.getProducts()));
+            case DATE -> input.setProducts(scoreSortDate(input.getProducts()));
```

Comment:
Wouldn't all of these fall through to the next case, since no break; is introduced?
So it would always set the Products based on `scoreSortDate`